lib/AST/Availability.cpp

```diff
@@ -749,3 +751,31 @@ ASTContext::getSwift5PlusAvailability(llvm::VersionTuple swiftVersion) {
 bool ASTContext::supportsVersionedAvailability() const {
   return minimumAvailableOSVersionForTriple(LangOpts.Target).has_value();
 }
+
+const Decl *
+swift::abstractSyntaxDeclForAvailableAttribute(const Decl *ConcreteSyntaxDecl) {
```

Comment:
Interesting, it feels like there's some overlap here with the `parentDeclForInferredAvailability` function that I introduced. I don't know that the two can merge, but maybe `parentDeclForInferredAvailability` should call this?